Add phase_order rake task and update README (#123)
procore/handcuffs: README.md

end
```

The default phase order in this case is determined by [Tsort](https://github.com/ruby/tsort) (topilogical sort). In order to validate the configuration and expected phase order it is reccomended that you check the phase configuration after any changes using the rake task:

```ruby
rake handcuffs:phase_order
```

This will display the default order in which phases will be run and list the pre-requisites of each phase. It will raise an error if there are any circular dependencies or if any pre-requisite is not a valid phase name.

### Tagging Migrations

Once configured, you can assign each migration to one of the defined phases using the `phase` setter method:

```ruby
# db/migrate/20240318230933_add_on_sale_column.rb

class AddOnSaleColumn < ActiveRecord::Migration[7.0]

  phase :pre_restart

  def change
    add_column :products, :on_sale, :boolean
  end
end
```

```ruby
# db/migrate/20240318230988_add_on_sale_index

class AddOnSaleIndex < ActiveRecord::Migration[7.0]

  phase :post_restart

  def change
    add_index :products, :on_sale, algorithm: :concurrently
  end
end
```

### Running Migrations In Phases

> After Handcuffs is configured and migrations are properly tagged, you can then run  migrations in phases using the `handcuffs:migrate` rake task with the specific phase to be run:

Reviewer: ```suggestion
After Handcuffs is configured and migrations are properly tagged, you can then run migrations in phases using the `handcuffs:migrate` rake task with the specific phase to be run:
```

```bash
rake 'handcuffs:migrate[pre_restart]'
```

or

```bash
rake 'handcuffs:migrate[post_restart]'
```

> *Note:* If you run phases out of order there are any pre-requisite phases (e.g. `phase :pre_restart`) with migrations that have not yet been run. In which case, trying to run a dependent phase (e.g. `phase: post_restart`) will raise a `HandcuffsPhaseOutOfOrderError`.

Reviewer: ```suggestion
*Note:* If you run phases out of order, or attempt to run a phase before outstanding migrations with a prerequisite phase have been run, a `HandcuffsPhaseOutOfOrderError` will be raised. 
```
